### ISSUE TITLE
`NoMethodError` in `DelegatorInspector` when #__getobj__ is protected

### ISSUE THREAD

**Issue by jtzero** [rspec/rspec-support#368]
### DelegatorInspector when :__getobj__ is  protected
```
NoMethodError:
       protected method `__getobj__' called for ......
```

### Your environment
* all
* rspec-support 3.8.0

### Steps to reproduce
```
describe 'DelegatorInspector error' do

  describe 'Axiom' do
    class ArgAxiom < SimpleDelegator
    end
    it 'will pass' do
      arg = ArgAxiom.new({hmmm: 'asdf'})
      expect(arg).to have_attributes(any?: true)
    end

    it 'is expected to error' do
      arg = ArgAxiom.new({hmmm: 'asdf'})
      expect(arg).to have_attributes(hmmm: true)
    end
  end

  describe 'Protected' do

    class ArgProtected < SimpleDelegator
      protected :__getobj__
    end
    it 'shouold not error' do
      arg = ArgProtected.new({})
      expect(arg).to have_attributes(any?: true)
    end
    it 'is expected to error' do
      arg = ArgProtected.new({})
      expect(arg).to have_attributes(any?: true)
    end
  end

end
```

### Expected behavior
the following errors shouldn't happen
```
SystemStackError:
       stack level too deep
```
```
NoMethodError:
       protected method `__getobj__' called for {}:ArgProtected
```

all of the errors should look like the Axiom failure
```
Failure/Error: expect(arg).to have_attributes(hmmm: true)
       expected #<ArgAxiom({:hmmm=>"asdf"})> to respond to :hmmm with 0 arguments
```

### Actual behavior
```
.FFFFF

Failures:

  1) DelegatorInspector error Axiom is expected to error
     Failure/Error: expect(arg).to have_attributes(hmmm: true)
       expected #<ArgAxiom({:hmmm=>"asdf"})> to respond to :hmmm with 0 arguments
     # ./tst_spec.rb:13:in `block (3 levels) in <top (required)>'
...
  4) DelegatorInspector error Protected shouold not error
     Failure/Error: expect(arg).to have_attributes(any?: true)
     
     NoMethodError:
       protected method `__getobj__' called for {}:ArgProtected
     # ./tst_spec.rb:40:in `block (3 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   protected method `__getobj__' called for {}:ArgProtected
     #   ./tst_spec.rb:40:in `block (3 levels) in <top (required)>'

  5) DelegatorInspector error Protected is expected to error
     Failure/Error: expect(arg).to have_attributes(any?: true)
     
     NoMethodError:
       protected method `__getobj__' called for {}:ArgProtected
     # ./tst_spec.rb:44:in `block (3 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   protected method `__getobj__' called for {}:ArgProtected
     #   ./tst_spec.rb:44:in `block (3 levels) in <top (required)>'

Finished in 0.01613 seconds (files took 0.08441 seconds to load)
6 examples, 5 failures

Failed examples:

rspec ./tst_spec.rb:11 # DelegatorInspector error Axiom is expected to error
...
rspec ./tst_spec.rb:38 # DelegatorInspector error Protected shouold not error
rspec ./tst_spec.rb:42 # DelegatorInspector error Protected is expected to error
```

edit: removed private references as that is not valid ruby

**Comment by JonRowe** [comment]
My apologies apparently we do call this for formatting purposes. Fixed by #369.

Please note it is still not a good idea to make this method protected/private... If you want a delegator without this magic I recommend implementing it yourself.

**Comment by jtzero** [comment]
No problem.

Please note I have little control over how libraries I don't contribute to implement their Delegators
